### DISCUSSION TITLE
Story Arc General Feedback

### DISCUSSION THREAD

**Discussion by Thom293** [AaronGullickson/mekhq#36] [General]
First - this is great.  Thank you.  Opens up so many possibilities.  I made it through Into the Storm so far.  Below is general feedback. 

If you would like me to create a separate issue for any of these, let me know and I will do so.

Forgive formatting.  This is a cut and paste from Excel.  Just thinking out loud feedback.  Feel free to ignore.

I   would consider this an Advanced Campagin.    Perhaps a note on the spash screen something to the effect of   "This is an advanced campaign and presumes you know how to fully use MHQ   and are familiar with the rules in BMM and Total Warfare and Campaign Ops and   …etc etc." Because you know a new player will start the suite and   immediatly go try to play the sexy Clan campaign and then have eleventy   billlion questions on discord.  It wont stop everyone, but maybe it will stop some.

- Char   Creation - just clicking on Leadership abilities (e.g. Tactics) charges your XP even before   you actually increase it. And then reduces again when you increase it.  I clicked away before I checked other   skills.  This seems to be a bug.
- Cutting   Teeth - Enemy starts with a damaged mech and only 14 units. So not 3 full   stars.  Maybe intentional.
- After   cutting Teeth - Should have basic Mass Repair/Salvage Settings pre-set? Maybe   not.
- Perhaps   the Tech should talk to us before battle to tell us to salvage mechs for   parts since parts are not buyable. I would have salvaged some other things   had I known.
- Hide   and Seek - I completed objectives and won battle, but at end of battle it   defaulted to "Draw" result.    Because I am foolish I didn’t check it and clicked done.  Then Natasha   told me I was a chump for doing poorly on the mission.
- Mess in   the mess hall.  I don’t know how to   rewind to read it, but I thought in 2nd to last paragraph the belligerents   have their names swapped, or the sexes are reversed earlier in the narrative,   or I just minsunderstood their portraits.    I thought the one that defended Phelan also lost.  But I could not double check because I   forgot to save before so dont have a way to rewind.  Perhaps a "back" button for   non-choice narrative panes?
- If the   narrative screen is up, clicking away from MHQ to another program and then   back to MHQ brings the narrative pane to the front, but not the rest of   MHQ.  Will post a screen as example.
- In the personell screen, injuried people show as red, but do not display their injury info on their person card on the right.  MHQ normally does.  Maybe intentional.
- One of   the Ryokyen have been checked as "total loss" at the end of each   battle.  Seems like the   "ghost" mech issue.  Not sure if   it was picked up during my save, or if it’s already part of the provided   Ryoken.  Im pretty sure it happened   after the first battle too, but not certain. 
- Question:  Once you complete arc, will you be allowed to contine in MHQ into a normal campaign?  Or does the game end?

Will try to play some more tomorrow and continue.  Once again, thank you.  This is great.

Screen showing only partial screen recovery of MHQ after clicking away.  Maybe just a MHQ bug, or maybe related to the Narrative panels.  Not sure.:

![image](https://github.com/AaronGullickson/mekhq/assets/23645569/0a59f38b-d5fe-4ee0-b242-140f93b6f365)

Screen Showing Ryoken as a total loss even though it was undamaged.  This may just be a MHQ bug, but it might be already applied the the Ryoken given, which is why I mention it here:

![image](https://github.com/AaronGullickson/mekhq/assets/23645569/16c4fb26-7010-4c25-ba52-ad82389a1a30)

**Comment by Thom293** [comment]
Also - Character Creation is awesome.  Would love it if it became part of the Company Generator and/or Normal MHQ new campaigns.

**Comment by AaronGullickson** [comment]
Thanks for these helpful comments. I am going to open discussions and move this over there. I will have more comments later.

**Comment by AaronGullickson** [comment]
I think some of these issues just reflect the campaign options. I should probably have a pop-up at the beginning explaining what options players are free to change and what should be left in place for the best experience. 

I noticed this problem as well with the Hide and Seek scenario. There are two issues really. The first is that this is a custom objective so the player has to check it. Really, it should be possible to designate hexes on the map in ScenarioObjectives and then check if they are destroyed, so we should work on implementing that. I think it was never implemented previously because with random maps in AtB, you cannot pre-designate. But you can if you specify a specific map.

The second issue is that even after manually checking the successful completion of the mission, the status does not change to victory. That seems like a proper bug and should be reported over at the official MekHQ GitHub page. 

I will check on the trothkin fight flow to see if something is going awry there. There are quite a few possible outcomes and the winner is determined randomly. Felipe is the warden and Ella is the crusader.

Not sure about the Ryoken total loss. I haven't seen that in my playthroughs and it sounds like it might be more a general MHQ bug. 

And yes, at the end of the arc, you will be left with your final campaign state and can continue to play.

**Comment by Thom293** [comment]
Just finished.  Great stuff!  I actually involuntarily blurted out "Hell yeah!" when I got to the end. That should be taken as a compliment.

I found 2 potential bugs, which I posted.  

Other random thoughts that occurred to me, which are just things to consider and can be ignored:

1.  Although I did not turn it on myself, someone told me there is a setting that prevents clan melee.  That might be a consideration for a default setting.  
2.  I dont know if its already possible, but having the ability to do narrative panes/pauses while on the MM map would allow for some mid-battle banter and choices.  Might be too much work though. Just an idea for later.

Thank you.  This can turn into something grand.  I have rough outlines of 2 story arcs that I did a bit ago.  They are a long ways off for me, but eventuallly I may bug you with a bunch of questions. :)

Thank you for doing this!  I love testing things, so if you need any help in that regard let me know. Congrats on this excellent work.

**Reply by AaronGullickson** [reply]
Thanks for testing it out, and congratulations on finishing. I see you went Crusader! One question: How did you find the Devin mission, Into the Storm? Just played it myself and realized that it is probably a bit too tough, so I toned down the number of units and how quickly they come.

I believe clan melee is being prevented, but the opfor is not properly being recognized as clan so they can physical but you can't which is not really fair. That is already fixed for next version.

I have thought about the idea of interrupting the MM game itself with narrative pauses, but that would be quite a project I think.  Something for the future possibly.

**Comment by Thom293** [comment]
I will be honest, I wanted the narrative and to see how the storyline worked, and I am not familiar with clan mechs.  I normally play IS only.  So, in each battle I gave my units to princess.  Its why I was able to finish so quickly.  I have messed with the AI settings before, and tested AI for clan specifically, so gave her some good ones, and she did just fine in Into the Storm.  (Best clan setting in my testing is Default v2 Braver.  But I turned herding up to 8 because I didnt want her to split up).  I think she lost me 2 mechs?  Given the OpFor, I think that is pretty good.   If you have suffered any attrition before that though, might make it rough.

She did get curbstomped on Ulric part II, but I think that would have been the outcome for me too.  Also it wasnt on the city map so there wasnt as much cover.   

I had typed this before related to Princess, but deleted it since it seems pretty tough on its own and doesnt need a difficulty bump, but I will type it again since you asked:

In the face of Clan firepower, Princess is very timid.  She will hang back at extreme range and just allow herself to get slowly picked off.  I think this is difficult on its own, so it probably does not need tweaking, but if you wanted to make it more difficult you could assign the Opfor Default v2 Braver.    I think that is the proper setting for Clan.  Its a little more aggressive but will still try to keep range.  Berserk v2 might actually be better in clan v clan.  But Berserk v2 is terrible for clans vs IS.  Again though - this seems difficult on its own so its probably not necessary for a change at all.

Also the Ulric parts on the city map - by default she will fall on her ass a lot on pavement and often kill many of her own in city maps before she even engages.  I did some work on that recently and she is a bit better, but its still imperfect.  Might want to consider a non-pavement non-swamp (still her two biggest issues) map for those battles.  But I didnt test that here specifically because I drew the forest maps.

--
Honestly since I made the changes to her fall risk, I need to fix Default v2 Braver to account for the new settings. It needs to be changed to the middle peg for fall risk now.  This has reminded me to do it.  Ill try to do it tomorrow night if I have time.   

I plan on playing out the battles next time I play it so I can learn to use clan mechs.  I especially want to do the group of 1v1 duels.  But I wanted to see the mechanics of the storyline first, so I cheated with Princess and fudged parts and repairs when I needed to do so.  Next time I will observe for difficulty and give feedback on that if necessary.  :)

**Reply by AaronGullickson** [reply]
This is very helpful. I think I will start an issue for tweaking the bot settings. I have noticed the tendency to fall back and avoid engaging in the default settings. Seems like you could mitigate the pavement problem by turning up the piloting caution, no?

**Reply by Thom293** [reply]
Pilot caution should be fixable by setting her caution to 5.  Default v2 Braver uses the old scale so is at 3.  Changing it to 5 should make her less disaster-prone.  I can try to test that Ancient City battle tonight.

**Comment by KaiserFalk** [comment]
I believe the xp thing for leadership has to do with how the xp table is set up. Some presets have unlocking a skill cost 0 XP, some have it cost XP (4XP in this preset)

I also noticed ghost damage in the hangar (marked as heavy damage/crippled when only missing a couple weapons)
